golang/networks/network_context.go

```diff
@@ -172,37 +161,50 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
 	}
 	resp, err := networkCtx.client.StartService(ctx, startServiceArgs)
 	if err != nil {
-		return nil, nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
+		return nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
 	}
 	logrus.Tracef("Successfully started service with Kurtosis API")
 
-	logrus.Tracef("Creating service interface...")
-	service := containerCreationConfig.GetServiceCreatingFunc()(serviceContext)
-	logrus.Tracef("Successfully created service interface")
-
-	networkCtx.services[serviceId] = service
-
-	availabilityChecker := services.NewDefaultAvailabilityChecker(serviceId, service)
-
-	return service, resp.UsedPortsHostPortBindings, availabilityChecker, nil
+	return serviceContext, resp.UsedPortsHostPortBindings,  nil
 }
 
 // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
-func (networkCtx *NetworkContext) GetService(serviceId services.ServiceID) (services.Service, error) {
-	networkCtx.mutex.Lock()
-	defer networkCtx.mutex.Unlock()
+func (networkCtx *NetworkContext) GetServiceContext(serviceId services.ServiceID, configFactory services.ContainerConfigFactory) (*services.ServiceContext, error) {
+	getServiceInfoArgs := &core_api_bindings.GetServiceInfoArgs{
+		ServiceId: string(serviceId),
+	}
+	serviceResponse, err := networkCtx.client.GetServiceInfo(context.Background(), getServiceInfoArgs)
+	if err != nil {
+		return nil, stacktrace.Propagate(
+			err,
+			"An error occurred when trying to get info from service '%v'",
```

Comment:
Very minor correction that doesn't really matter but figured I'd point it out for English-bettering purposes - this should "get info for service" rather than "get info from service", since we're actually getting info from the API container for the service